plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/RegisterAppEngineStandardRuntimeStartup.java

```diff
@@ -0,0 +1,49 @@
+
+package com.google.cloud.tools.eclipse.appengine.localserver.server;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.wst.server.core.IRuntime;
+import org.eclipse.wst.server.core.IRuntimeType;
+import org.eclipse.wst.server.core.IRuntimeWorkingCopy;
+import org.eclipse.wst.server.core.ServerCore;
+import org.eclipse.wst.server.core.internal.IStartup;
+
+/**
+ * Registers the default instance of our App Engine Standard runtime if not found. Uses the internal
+ * WTP {@link IStartup} to minimize activation to use of WTP-related functionality.
+ */
+public class RegisterAppEngineStandardRuntimeStartup implements IStartup {
+  private static final Logger logger =
+      Logger.getLogger(RegisterAppEngineStandardRuntimeStartup.class.getName());
+
+  /**
+   * AppEngineStandardFacet#createAppEngineServerRuntime() creates the default runtime using a
+   * {@code null} ID, which causes it to take the name of its runtime-type.
+   */
+  private static final String RUNTIME_ID = "App Engine Standard Runtime";
+
+  @Override
+  public void startup() {
+    IRuntime runtime = ServerCore.findRuntime(RUNTIME_ID);
```

Comment:
Why don't we just call the `static` method `createAppEngineServerRuntime()`? The code looks very similar. Even if there is some difference (I don't see a major one) here, I think it's better to have a single code place to create the runtime.

Reply:
I do remember considering that… but I don't know why I didn't follow through.